### PR TITLE
clang-tidy: ignore dynamic_modules/abi.h and test C programs

### PR DESCRIPTION
**Pull request by mathetake**
part of #28566

### CODE REVIEW

File: source/extensions/dynamic_modules/abi.h [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 // NOLINT(namespace-envoy)
```
Comment (mathetake) [inline source/extensions/dynamic_modules/abi.h:3]: this is anyway needed as this is checked in format, not clang-tidy

**Comment by mathetake** [comment]
cc @phlax

**Review by phlax — DISMISSED** [review]
lgtm, thanks @mathetake

**Comment by mathetake** [comment]
damn spelling

**Review by phlax — APPROVED** [review]

**Pull request merged**